internal/service/studies/agreements.go

```diff
@@ -0,0 +1,58 @@
+package studies
+
+import (
+	"github.com/google/uuid"
+	"github.com/ucl-arc-tre/portal/internal/config"
+	openapi "github.com/ucl-arc-tre/portal/internal/openapi/web"
+	"github.com/ucl-arc-tre/portal/internal/types"
+)
+
+// creates a signature record for a user confirming an agreement for a specific study
+func (s *Service) ConfirmStudyAgreement(user types.User, studyID uuid.UUID, agreementID uuid.UUID) error {
+	// Verify the study exists and the user has access to it
+	_, err := s.StudyWithOwner(studyID, user)
+	if err != nil {
+		return err
+	}
+
+	signature := types.StudyAgreementSignature{
+		UserID:      user.ID,
+		StudyID:     studyID,
+		AgreementID: agreementID,
+	}
+
+	result := s.db.Where(&signature).FirstOrCreate(&signature)
+	if result.Error != nil {
+		return types.NewErrServerError(result.Error)
+	}
+
+	return nil
+}
+
+// returns all agreement signatures for a specific study and user
+func (s *Service) GetStudyAgreementSignatures(user types.User, studyID uuid.UUID) ([]openapi.ConfirmedAgreement, error) {
+	// Verify the study exists and the user has access to it
+	_, err := s.StudyWithOwner(studyID, user)
+	if err != nil {
+		return nil, err
+	}
+
+	var studySignatures []types.StudyAgreementSignature
+	result := s.db.Preload("Agreement").
```

Comment:
I don't think we want to preload the agreement – will be a load of text(?)

Reply:
Yeah I was going back and forth on that one myself. I don't think it's being directly utilised so I might just get rid of the preload here.